akka-actor/src/main/scala/akka/pattern/RetrySupport.scala

```diff
@@ -282,6 +284,31 @@ trait RetrySupport {
       attempted = 0,
       retrySettings.shouldRetry)
   }
+
+  /**
+   * Given a function from Unit to Future, returns an internally retrying Future.
+   * The first attempt will be made immediately, any subsequent attempt will be made based on provided [[RetrySettings]].
+   *
+   * <b>Example usage:</b>
+   *
+   * // retry with backoff
+   * {{{
+   * protected val sendAndReceive: HttpRequest => Future[HttpResponse] = { (req) => ??? }
+   * private val sendReceiveRetry: HttpRequest => Future[HttpResponse] = (req: HttpRequest) => retry[HttpResponse](
+   *   RetrySettings(10)) {
+   *      () => sendAndReceive(req)
+   *   }
+   * }}}
+   */
+  def retry[T](retrySettings: RetrySettings, system: ClassicActorSystemProvider)(
+      attempt: () => Future[T]): Future[T] = {
+    RetrySupport.retry(
+      attempt,
+      retrySettings.maxRetries,
+      retrySettings.delayFunction,
+      attempted = 0,
+      retrySettings.shouldRetry)(system.classicSystem.dispatcher, system.classicSystem.scheduler)
+  }
```

Comment:
Add a test in akka-actor-typed tests that it can actually be used

Reply:
Too quick, we want it implicit (so it will conflict with the other one, but just having this one is fine)